proxy/src/telemetry/metrics/serve.rs

```diff
@@ -27,29 +40,29 @@ impl Serve {
         }
     }
 
-    fn is_gzip(req: &Request) -> bool {
-        if let Some(accept_encodings) = req
-            .headers()
-            .get::<AcceptEncoding>()
-        {
-            return accept_encodings
-                .iter()
-                .any(|&QualityItem { ref item, .. }| item == &Encoding::Gzip)
-        }
-        false
+    fn is_gzip<B>(req: &Request<B>) -> bool {
+        req.headers()
+            .get_all(header::ACCEPT_ENCODING).iter()
+            .any(|value| {
+                value.to_str().ok()
+                    .map(|value| value.contains("gzip"))
+                    .unwrap_or(false)
+            })
     }
 }
 
 impl Service for Serve {
-    type Request = Request;
-    type Response = Response;
-    type Error = hyper::Error;
-    type Future = FutureResult<Self::Response, Self::Error>;
-
-    fn call(&self, req: Self::Request) -> Self::Future {
-        if req.path() != "/metrics" {
-            return future::ok(Response::new()
-                .with_status(StatusCode::NotFound));
+    type ReqBody = Body;
+    type ResBody = Body;
+    type Error = ServeError;
+    type Future = FutureResult<Response<Body>, Self::Error>;
+
+    fn call(&mut self, req: Request<Body>) -> Self::Future {
+        if req.uri().path() != "/metrics" {
+            return future::result(Response::builder()
+                .status(StatusCode::NOT_FOUND)
+                .body(Body::empty())
+                .map_err(ServeError::from));
```

Comment:
This builder shouldn't error, since status was called with an existing `StatusCode`.